HeartbeatLogging/Sources/HeartbeatStorage.swift

```diff
@@ -14,31 +14,44 @@
 
 import Foundation
 
+/// A typealias for a function that transforms a `HeartbeatInfo?` into `HeartbeatInfo?`. This
+/// typealias is used in `HeartbeatStorageProtocol` to model the transformation of the contents
+/// of heartbeat storage.
+///
+/// The parameter is marked as optional as the heartbeat info provided to the closure may be `nil` if
+/// an error occurred or storage is empty. The return type is marked as optional because the storage can
+/// be emptied by returning `nil` from the transformation closure.
 typealias HeartbeatInfoTransform = (HeartbeatInfo?) -> HeartbeatInfo?
```

Comment:
Consider eliminating the typealias as we discussed.

Reply:
Reworked in 861028b. Im going to followup with some other improvements that should further simplify things